### PR TITLE
[TUTORIAL] Add matrix vector multiplication tutorial

### PR DESCRIPTION
**Pull request by Kuigesi**

### CODE REVIEW

File: python/tutorials/matrix-vector-multiplication.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+LINE_VALS = ['triton-cpu-single', 'triton-cpu', 'torch-cpu-native', 'torch-cpu-compile']
+LINE_NAMES = ['TritonCPU 1', 'TritonCPU', 'TorchCPU (native)', 'TorchCPU (compile)']
+LINE_STYLES = [('blue', '--'), ('blue', '-'), ('green', '--'), ('green', '-')]
+
+if USE_GPU and triton.runtime.driver.get_active_gpus():
+    triton.runtime.driver.set_active_to_gpu()
+    weight = weight.to('cuda')
+    x = x.to('cuda')
+    triton_output = gemv(weight, x, None)
```
Comment (embg) [inline python/tutorials/matrix-vector-multiplication.py:101]: Does Triton decide whether to call CPU or GPU kernel based on which device holds the tensor?

Comment (Kuigesi) [reply python/tutorials/matrix-vector-multiplication.py:101]: I believe it is controled by line 98: triton.runtime.driver.set_active_to_gpu(), it changes the backend driver

File: python/tutorials/matrix-vector-multiplication.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
+import torch
+
+import triton
+import triton.language as tl
+
+BLOCK_M = 1
+BLOCK_N = 512
+USE_GPU = False
+"""
+Kernel for computing Y = A @ X, where A is a dense matrix with
```
Comment (digantdesai) [inline python/tutorials/matrix-vector-multiplication.py:10]: typically it is MxK @ KxN = MxK where we reduce over K-dim.

File: python/tutorials/matrix-vector-multiplication.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
+import torch
+
+import triton
+import triton.language as tl
+
+BLOCK_M = 1
```
Comment (digantdesai) [inline python/tutorials/matrix-vector-multiplication.py:6]: nit
```suggestion
BLOCK_SIZE_M = 1
```

File: python/tutorials/matrix-vector-multiplication.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    rn = tl.arange(0, BLOCK_N)
+
+    A = A + (rm[:, None] * stride_am + rn[None, :])
+    X = X + rn
+
+    acc = tl.zeros((BLOCK_M, ), dtype=tl.float32)
+    for n in range(N, 0, -BLOCK_N):
+        a = tl.load(A)
+        x = tl.load(X)
+        acc += tl.sum(a * x[None, :], 1)
```
Comment (digantdesai) [inline python/tutorials/matrix-vector-multiplication.py:40]:
```suggestion
        acc += tl.sum(a * x[None, :], axis=1)
```

File: python/tutorials/matrix-vector-multiplication.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        ms, min_ms, max_ms = triton.testing.do_bench(lambda: gemv(weight, x, output), quantiles=quantiles)
+    elif provider == 'torch-cpu-native':
+        ms, min_ms, max_ms = triton.testing.do_bench(lambda: torch.matmul(weight, x, out=output), quantiles=quantiles,
+                                                     is_cpu=True)
+    elif provider == 'torch-cpu-compile':
+        compiled = torch.compile(torch.matmul)
+        ms, min_ms, max_ms = triton.testing.do_bench(lambda: compiled(weight, x, out=output), quantiles=quantiles,
+                                                     is_cpu=True)
+    elif provider == 'triton-cpu-single':
+        ms, min_ms, max_ms = triton.testing.do_bench(lambda: gemv(weight, x, output), quantiles=quantiles, is_cpu=True)
```
Comment (digantdesai) [inline python/tutorials/matrix-vector-multiplication.py:169]: how are we setting OMP num threads = 1?

Comment (Kuigesi) [reply python/tutorials/matrix-vector-multiplication.py:169]: In line 149, we are setting the env variable, os.environ['TRITON_CPU_SINGLE_CORE'] = '1'

File: python/tutorials/matrix-vector-multiplication.py (hunk 2 above)
Comment (Kuigesi) [reply python/tutorials/matrix-vector-multiplication.py:10]: I followed the naming convention in https://github.com/FMInference/DejaVu/blob/master/Dejavu/src/ops/triton/gather_gemv.py.

I think for gemv, people also use M,N instead of M,K, see https://oneapi-src.github.io/oneMKL/domains/blas/gemv.html

File: python/tutorials/matrix-vector-multiplication.py (hunk 3 above)
Comment (Kuigesi) [reply python/tutorials/matrix-vector-multiplication.py]: Fixed

File: python/tutorials/matrix-vector-multiplication.py (hunk 4 above)
Comment (Kuigesi) [reply python/tutorials/matrix-vector-multiplication.py]: fixed

File: python/tutorials/matrix-vector-multiplication.py (hunk 2 above)
Comment (digantdesai) [reply python/tutorials/matrix-vector-multiplication.py:10]: TIL. Thanks.

**Review by minjang — APPROVED** [review]
LGTM, resolved all the comments.
